openfaas/nats-queue-worker
# Moved vendoring tool to dep.

Signed-off-by: dungeonmaster18 <umesh4257@gmail.com>

**Description:**
This pr moves this repo to dep from vndr.

**Test**:
* First I did ```dep init``` on the entire repo.
* build the the docker image for worker-queue
* Now cloned the faas repo and changed the functions/queue-worker:0.4 to functions/queue-worker:latest-dev and then run ```./deploy-stack.sh```
* To test my changes were successful or not I did following things
   
* I called the function with async
    
```
$ curl -v localhost:8080/async-function/func_echoit/ -d "You should see HTTP/1.1 202 Accepted as expected"
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8080 (#0)
> POST /async-function/func_echoit/ HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.57.0
> Accept: */*
> Content-Length: 48
> Content-Type: application/x-www-form-urlencoded
>
* upload completely sent off: 48 out of 48 bytes
< HTTP/1.1 202 Accepted
< Date: Sat, 20 Jan 2018 14:45:23 GMT
< Content-Length: 0
< Content-Type: text/plain; charset=utf-8
<
* Connection #0 to host localhost left intact
```
  * Then checked worker-log
 ```
$ docker service logs -f func_queue-worker
func_queue-worker.1.v114itacm46k@node2    | Wait for  30s
func_queue-worker.1.v114itacm46k@node2    | Listening on [faas-request], clientID=[faas-worker-433394d71e07], qgroup=[faas] durable=[]
func_queue-worker.1.v114itacm46k@node2    | [#1] Received on [faas-request]: 'sequence:7 subject:"faas-request" data:"{\"Header\":{\"Accept\":[\"*/*\"],\"Content-Length\":[\"48\"],\"Content-Type\":[\"application/x-www-form-urlencoded\"],\"User-Agent\":[\"curl/7.57.0\"]},\"Body\":\"WW91IHNob3VsZCBzZWUgSFRUUC8xLjEgMjAyIEFjY2VwdGVkIGFzIGV4cGVjdGVk\",\"Method\":\"POST\",\"QueryString\":\"\",\"Function\":\"func_echoit\",\"CallbackUrl\":null}" timestamp:1516459349333418594 '
func_queue-worker.1.v114itacm46k@node2    | Request for func_echoit.
func_queue-worker.1.v114itacm46k@node2    | You should see HTTP/1.1 202 Accepted as expected
func_queue-worker.1.v114itacm46k@node2    | 200 OK
func_queue-worker.1.v114itacm46k@node2    | Posting report - 200
func_queue-worker.1.v114itacm46k@node2    | [#2] Received on [faas-request]: 'sequence:8 subject:"faas-request" data:"{\"Header\":{\"Accept\":[\"*/*\"],\"Content-Length\":[\"48\"],\"Content-Type\":[\"application/x-www-form-urlencoded\"],\"User-Agent\":[\"curl/7.57.0\"]},\"Body\":\"WW91IHNob3VsZCBzZWUgSFRUUC8xLjEgMjAyIEFjY2VwdGVkIGFzIGV4cGVjdGVk\",\"Method\":\"POST\",\"QueryString\":\"\",\"Function\":\"func_echoit\",\"CallbackUrl\":null}" timestamp:1516459523135862335 '
func_queue-worker.1.v114itacm46k@node2    | Request for func_echoit.
func_queue-worker.1.v114itacm46k@node2    | You should see HTTP/1.1 202 Accepted as expected
func_queue-worker.1.v114itacm46k@node2    | 200 OK
func_queue-worker.1.v114itacm46k@node2    | Posting report - 200
```

* Also tested with the requestbin.in
```
$ curl localhost:8080/async-function/func_echoit/ -d "test requestbin without params" -H "X-Callback-Url: https://requestb.in/yozjb8yp"
$ curl localhost:8080/async-function/func_echoit/ -d "test requestbin params" -H "X-Callback-Url: https://requestb.in/yozjb8yp?faas=isworking"
$ docker service logs -f func_queue-workerfunc_queue-worker.1.v114itacm46k@node2    | Wait for  30s
func_queue-worker.1.v114itacm46k@node2    | Listening on [faas-request], clientID=[faas-worker-433394d71e07], qgroup=[faas] durable=[]
func_queue-worker.1.v114itacm46k@node2    | [#1] Received on [faas-request]: 'sequence:7 subject:"faas-request" data:"{\"Header\":{\"Accept\":[\"*/*\"],\"Content-Length\":[\"48\"],\"Content-Type\":[\"application/x-www-form-urlencoded\"],\"User-Agent\":[\"curl/7.57.0\"]},\"Body\":\"WW91IHNob3VsZCBzZWUgSFRUUC8xLjEgMjAyIEFjY2VwdGVkIGFzIGV4cGVjdGVk\",\"Method\":\"POST\",\"QueryString\":\"\",\"Function\":\"func_echoit\",\"CallbackUrl\":null}" timestamp:1516459349333418594 '
func_queue-worker.1.v114itacm46k@node2    | Request for func_echoit.
func_queue-worker.1.v114itacm46k@node2    | You should see HTTP/1.1 202 Accepted as expected
func_queue-worker.1.v114itacm46k@node2    | 200 OK
func_queue-worker.1.v114itacm46k@node2    | Posting report - 200
func_queue-worker.1.v114itacm46k@node2    | [#2] Received on [faas-request]: 'sequence:8 subject:"faas-request" data:"{\"Header\":{\"Accept\":[\"*/*\"],\"Content-Length\":[\"48\"],\"Content-Type\":[\"application/x-www-form-urlencoded\"],\"User-Agent\":[\"curl/7.57.0\"]},\"Body\":\"WW91IHNob3VsZCBzZWUgSFRUUC8xLjEgMjAyIEFjY2VwdGVkIGFzIGV4cGVjdGVk\",\"Method\":\"POST\",\"QueryString\":\"\",\"Function\":\"func_echoit\",\"CallbackUrl\":null}" timestamp:1516459523135862335 '
func_queue-worker.1.v114itacm46k@node2    | Request for func_echoit.
func_queue-worker.1.v114itacm46k@node2    | You should see HTTP/1.1 202 Accepted as expected
func_queue-worker.1.v114itacm46k@node2    | 200 OK
func_queue-worker.1.v114itacm46k@node2    | Posting report - 200
func_queue-worker.1.v114itacm46k@node2    | [#3] Received on [faas-request]: 'sequence:9 subject:"faas-request" data:"{\"Header\":{\"Accept\":[\"*/*\"],\"Content-Length\":[\"30\"],\"Content-Type\":[\"application/x-www-form-urlencoded\"],\"User-Agent\":[\"curl/7.57.0\"],\"X-Callback-Url\":[\"https://requestb.in/yozjb8yp\"]},\"Body\":\"dGVzdCByZXF1ZXN0YmluIHdpdGhvdXQgcGFyYW1z\",\"Method\":\"POST\",\"QueryString\":\"\",\"Function\":\"func_echoit\",\"CallbackUrl\":{\"Scheme\":\"https\",\"Opaque\":\"\",\"User\":null,\"Host\":\"requestb.in\",\"Path\":\"/yozjb8yp\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"}}" timestamp:1516459584037338014 '
func_queue-worker.1.v114itacm46k@node2    | Request for func_echoit.
func_queue-worker.1.v114itacm46k@node2    | test requestbin without params
func_queue-worker.1.v114itacm46k@node2    | 200 OK
func_queue-worker.1.v114itacm46k@node2    | Callback to: https://requestb.in/yozjb8yp
func_queue-worker.1.v114itacm46k@node2    | Posted result: 200
func_queue-worker.1.v114itacm46k@node2    | Posting report - 200
func_queue-worker.1.v114itacm46k@node2    | Request for func_echoit.
func_queue-worker.1.v114itacm46k@node2    | [#4] Received on [faas-request]: 'sequence:10 subject:"faas-request" data:"{\"Header\":{\"Accept\":[\"*/*\"],\"Content-Length\":[\"22\"],\"Content-Type\":[\"application/x-www-form-urlencoded\"],\"User-Agent\":[\"curl/7.57.0\"],\"X-Callback-Url\":[\"https://requestb.in/yozjb8yp?faas=isworking\"]},\"Body\":\"dGVzdCByZXF1ZXN0YmluIHBhcmFtcw==\",\"Method\":\"POST\",\"QueryString\":\"\",\"Function\":\"func_echoit\",\"CallbackUrl\":{\"Scheme\":\"https\",\"Opaque\":\"\",\"User\":null,\"Host\":\"requestb.in\",\"Path\":\"/yozjb8yp\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"faas=isworking\",\"Fragment\":\"\"}}" timestamp:1516459658340734327 '
func_queue-worker.1.v114itacm46k@node2    | test requestbin params
func_queue-worker.1.v114itacm46k@node2    | 200 OK
func_queue-worker.1.v114itacm46k@node2    | Callback to: https://requestb.in/yozjb8yp?faas=isworking
func_queue-worker.1.v114itacm46k@node2    | Posted result: 200
func_queue-worker.1.v114itacm46k@node2    | Posting report - 200
```

CC / @stefanprodan @alexellis
